music21/metadata/__init__.py

```diff
@@ -2605,6 +2554,185 @@ def update(self, streamObj):
+         ('software', 'Finale 2014 for Mac'),
+         ('software', 'music21 v.8...'),
+         ('sourcePath', 'corelli/opus3no1/1grave.xml'),
+         ('tempoFirst', None),
+         ('tempos', []),
+         ('timeSignatureFirst', '4/4'),
+         ('timeSignatures', ['4/4']))
+        '''
+        allOut: t.List[t.Tuple[str, t.Any]] = list(super().all(skipContributors))
+
+        for name in self.additionalRichMetadataAttributes:
+            allOut.append((name, getattr(self, name)))
+
+        return tuple(sorted(allOut))
+
+    def getAllNamedValues(self, skipContributors=False) -> t.Tuple[t.Tuple[str, ValueType], ...]:
+        '''
+        Returns all values stored in this RichMetadata as a tuple of (name, value) tuples.
+        Names with multiple values will appear multiple times in the list (rather
+        than appearing once, with a value that is a list of values).
+        The tuple's first element (the name) is either of the form 'namespace:name', or a
+        custom name (with no form at all).
+
+        >>> rmd = metadata.RichMetadata()
+        >>> c = corpus.parse('corelli/opus3no1/1grave')
+        >>> rmd.merge(c.metadata)
+        >>> rmd.update(c)
+        >>> all = rmd.getAllNamedValues()
+        >>> all
+        (('musicxml:software', <music21.metadata.primitives.Text music21 v.8...>),
+         ('musicxml:software', <music21.metadata.primitives.Text Finale 2014 for Mac>),
+         ('musicxml:software', <...Text Dolet Light for Finale 2014>),
+         ('humdrum:OMD', <...Text Sonata da Chiesa, No. I (opus 3, no. 1)>),
+         ('marcrel:CMP', <...Contributor composer:Arcangelo Corelli>),
+         ('marcrel:ARR',  <...Contributor arranger:Michael Scott Cuthbert>),
+         ('dcterms:rights', <...Copyright © 2014, Creative Commons License (CC-BY)>),
+         ('m21FileInfo:filePath', <...Text ...corpus/corelli/opus3no1/1grave.xml>),
+         ('m21FileInfo:fileFormat', <music21.metadata.primitives.Text musicxml>),
+         ('ambitus',
+            AmbitusShort(semitones=48, diatonic='P1', pitchLowest='C2', pitchHighest='C6')),
+         ('keySignatureFirst', -1),
+         ('keySignatures', [-1]),
+         ('noteCount', 259),
+         ('numberOfParts', 3),
+         ('pitchHighest', 'C6'),
+         ('pitchLowest', 'C2'),
+         ('quarterLength', 76.0),
+         ('sourcePath', 'corelli/opus3no1/1grave.xml'),
+         ('tempoFirst', None),
+         ('tempos', []),
+         ('timeSignatureFirst', '4/4'),
+         ('timeSignatures', ['4/4']))
+
+        >>> allNonContributors = rmd.getAllNamedValues(skipContributors=True)
+        >>> allNonContributors
+        (('musicxml:software', <music21.metadata.primitives.Text music21 v.8...>),
+         ('musicxml:software', <music21.metadata.primitives.Text Finale 2014 for Mac>),
+         ('musicxml:software', <...Text Dolet Light for Finale 2014>),
+         ('humdrum:OMD', <...Text Sonata da Chiesa, No. I (opus 3, no. 1)>),
```

Comment:
as above, don't duplicate everything to show a few changes.  it's hard to see that the composers, etc. are missing.

Reply:
Agreed.